platform/admin/admin-configuration-configupdater/src/main/java/org/codice/ddf/admin/configuration/ConfigurationUpdater.java

```diff
@@ -0,0 +1,322 @@
+ * enableConfigSave} in {@code custom.properties} should be set to {@code false}.
+ */
+public class ConfigurationUpdater implements ConfigurationPersistencePlugin {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationUpdater.class);
+
+  private static final Pattern ENC_PATTERN = Pattern.compile("^ENC\\((.*)\\)$");
+
+  static final String FELIX_FILENAME = "felix.fileinstall.filename";
+
+  private final ConfigurationAdmin ddfConfigAdmin;
+
+  private final List<PersistenceStrategy> strategies;
+
+  private final EncryptionService encryptionService;
+
+  private final Map<String, CachedConfigData> pidDataMap;
+
+  /*
+   * SortedServiceList implements java.util.List using generic parameter <T> and not a concrete type,
+   * so the ctor cannot be matched by the blueprint container if we did List<PersistenceStrategy>
+   *
+   * See https://issues.apache.org/jira/browse/ARIES-960
+   */
+  public ConfigurationUpdater(
+      ConfigurationAdmin ddfConfigAdmin, List strategies, EncryptionService encryptionService) {
+    this.ddfConfigAdmin = ddfConfigAdmin;
+    this.strategies = strategies;
+    this.encryptionService = encryptionService;
+    this.pidDataMap = new ConcurrentHashMap<>();
+  }
+
+  /** @return a read-only map of configuration pids to felix file install properties. */
+  @VisibleForTesting
+  Map<String, CachedConfigData> getPidDataMap() {
+    return Collections.unmodifiableMap(pidDataMap);
+  }
+
+  /**
+   * @inheritDoc
+   * @throws IOException if an error occurs reading configuration data
+   * @throws InvalidSyntaxException technically impossible since {@code null} is being passed as the
+   *     configuration filter. See {@link ConfigurationAdmin#listConfigurations(String)}.
+   * @see ConfigurationPersistencePlugin
+   */
+  @Override
+  public void initialize(ConfigurationContextFactory factory) {
+    Configuration[] configs;
+    try {
+      configs = ddfConfigAdmin.listConfigurations(null);
+    } catch (IOException | InvalidSyntaxException e) {
+      LOGGER.error("Problem fetching configurations. ", e);
+      return;
+    }
+    if (configs == null) {
+      return;
+    }
+    pidDataMap.putAll(
+        Arrays.stream(configs)
+            .map(factory::createContext)
+            .filter(context -> context.getConfigFile() != null)
+            .collect(Collectors.toMap(ConfigurationContext::getServicePid, CachedConfigData::new)));
+  }
+
+  @Override
+  public void handleStore(ConfigurationContext context) throws IOException {
+    try {
+      AccessController.doPrivileged(
+          (PrivilegedExceptionAction<Object>)
+              () -> {
+                doHandleStore(context);
+                return null;
+              });
+    } catch (PrivilegedActionException e) {
+      // Only caught if the action threw a CHECKED exception, thus it MUST be an IOException.
+      throw (IOException) e.getCause();
```

Comment:
❗️ As per a previous comment, we shouldn't blindly cast even if we know by design these are the only exceptions that are thrown. That side can evolve differently and we would then have a problem here.